shared-module/src/components/InputFields/TextField.tsx

```diff
@@ -73,11 +51,37 @@ const errorClass = css`
   display: inline-block;
 `
 
-export type TextFieldProps = Omit<React.HTMLAttributes<HTMLInputElement>, "onChange"> &
-  TextFieldExtraProps
+export interface TextFieldProps extends InputHTMLAttributes<HTMLInputElement> {
+  type?: "email" | "password" | "text" | "number" | "color"
+  label?: string
+  labelStyle?: string
+  hint?: string
+  error?: string | FieldError
+  register?: UseFormRegisterReturn
+  onChangeByValue?: (value: string, name?: string) => void
+}
 
-const TextField = ({ onChange, className, register, disabled, error, ...rest }: TextFieldProps) => {
+const TextField: React.FC<TextFieldProps> = ({
+  onChange,
+  onChangeByValue,
+  className,
+  register,
+  disabled,
+  error,
+  ...rest
+}: TextFieldProps) => {
   const { t } = useTranslation()
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (onChangeByValue) {
+      const {
+        target: { value },
+      } = event
+      onChangeByValue(value)
+    } else if (onChange) {
```

Comment:
make this a regular if so that if someone provides both onChangeByValue and onChange, both would get run